win32ss/user/ntuser/window.c

```diff
@@ -1218,29 +1218,29 @@ co_IntSetParent(PWND Wnd, PWND WndNewParent)
 
    if (WndOldParent) UserReferenceObject(WndOldParent); /* Caller must deref */
 
-   if (WndNewParent != WndOldParent)
-   {
-      /* Unlink the window from the siblings list */
-      IntUnlinkWindow(Wnd);
-      Wnd->ExStyle2 &= ~WS_EX2_LINKED;
+   /* Even if WndNewParent == WndOldParent continue because the
+    * child window (Wnd) should be moved to the top of the z-order */
 
-      /* Set the new parent */
-      WndSetParent(Wnd, WndNewParent);
+   /* Unlink the window from the siblings list */
+   IntUnlinkWindow(Wnd);
+   Wnd->ExStyle2 &= ~WS_EX2_LINKED;
 
-      if ( Wnd->style & WS_CHILD &&
-           Wnd->spwndOwner &&
-           Wnd->spwndOwner->ExStyle & WS_EX_TOPMOST )
-      {
-         ERR("SetParent Top Most from Pop up!\n");
-         Wnd->ExStyle |= WS_EX_TOPMOST;
-      }
+   /* Set the new parent */
+   WndSetParent(Wnd, WndNewParent);
```

Comment:
Keep the existing `if()` around this,
or add an(other) explicit comment that call is useless when New is equal to Old.

Reply:
does it actually hurt in any way if that if is not present? Can you come up with any specific failure if it is missing?
Otherwise why not sticking with the simpler code?

Reply:
> does it actually hurt in any way if that if is not present?

Not with current code: only useless call, and a bit misleading (comment).

> Otherwise why not sticking with the simpler code?

My suggestion: minimal change, or update comment.